packages/web-components/src/text-input/text-input.spec.ts

```diff
@@ -0,0 +1,636 @@
+                <fluent-text-input autofocus></fluent-text-input>
+            `;
+    });
+
+    await expect(control).toHaveAttribute('autofocus', '');
+  });
+
+  test('should set the `disabled` attribute on the internal control', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input disabled></fluent-text-input>
+            `;
+    });
+
+    await expect(control).toHaveAttribute('disabled', '');
+  });
+
+  test('should set the `readonly` attribute on the internal control', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input readonly></fluent-text-input>
+            `;
+    });
+
+    await expect(control).toHaveAttribute('readonly', '');
+  });
+
+  test('should set the `required` attribute on the internal control', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input required></fluent-text-input>
+            `;
+    });
+
+    await expect(control).toHaveAttribute('required', '');
+  });
+
+  test('should set the `spellcheck` attribute on the internal control', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input spellcheck></fluent-text-input>
+            `;
+    });
+
+    await expect(control).toHaveAttribute('spellcheck', '');
+  });
+
+  test.describe('should set the attribute on the internal control', () => {
+    const attributes = {
+      maxlength: 14,
+      minlength: 14,
+      name: 'foo',
+      placeholder: 'foo',
+      size: 4,
+      list: 'listId',
+      ariaAtomic: 'true',
+      ariaBusy: 'false',
+      ariaControls: 'testId',
+      ariaCurrent: 'page',
+      ariaDescribedby: 'testId',
+      ariaDetails: 'testId',
+      ariaDisabled: 'true',
+      ariaErrormessage: 'test',
+      ariaFlowto: 'testId',
+      ariaHaspopup: 'true',
+      ariaHidden: 'true',
+      ariaInvalid: 'spelling',
+      ariaKeyshortcuts: 'F4',
+      ariaLabel: 'Foo label',
+      ariaLabelledby: 'testId',
+      ariaLive: 'polite',
+      ariaOwns: 'testId',
+      ariaRelevant: 'removals',
+      ariaRoledescription: 'search',
+    };
+
+    for (const [attribute, value] of Object.entries(attributes)) {
+      const attrToken = spinalCase(attribute);
+
+      test(`should set the \`${attrToken}\` attribute on the internal control`, async () => {
+        await root.evaluate(
+          (node, { attrToken, value }) => {
+            node.innerHTML = /* html */ `
+                            <fluent-text-input ${attrToken}="${value}"></fluent-text-input>
+                        `;
+          },
+          { attrToken, value },
+        );
+
+        await expect(control).toHaveAttribute(attrToken, `${value}`);
+      });
+    }
+  });
+  test('should reflect control-size attribute values', async () => {
+    await element.evaluate((node: TextInput) => {
```

Comment:
Why don't we also test how `<fluent-text-input contol-size="small">` renders?
I understand the attribute - JS reflection is enough, but still would like to see the "declarative render".